src/cursor.c

```diff
@@ -344,50 +361,8 @@ process_cursor_motion(struct server *server, uint32_t time)
 		}
 	}
 
-	/* TODO: ssd_hover_state should likely be located in server->seat */
-	ssd_update_button_hover(node, &server->ssd_hover_state);
-
-	if (server->seat.pressed.surface &&
-			server->seat.pressed.surface != surface &&
-			!server->seat.drag_icon) {
-		/*
-		 * Button has been pressed while over another surface
-		 * and is still held down. Just send the adjusted motion
-		 * events to the focused surface so we can keep scrolling
-		 * or selecting text even if the cursor moves outside of
-		 * the surface.
-		 */
-		process_cursor_motion_out_of_surface(server, time);
-	} else if (surface && !input_inhibit_blocks_surface(
-			&server->seat, surface->resource)) {
-		bool focus_changed =
-			wlr_seat->pointer_state.focused_surface != surface;
-		/*
-		 * "Enter" the surface if necessary. This lets the client know
-		 * that the cursor has entered one of its surfaces.
-		 *
-		 * Note that this gives the surface "pointer focus", which is
-		 * distinct from keyboard focus. You get pointer focus by moving
-		 * the pointer over a window.
-		 */
-		wlr_seat_pointer_notify_enter(wlr_seat, surface, sx, sy);
-		if (!focus_changed || server->seat.drag_icon) {
-			/*
-			 * The enter event contains coordinates, so we only need
-			 * to notify on motion if the focus did not change.
-			 */
-			wlr_seat_pointer_notify_motion(wlr_seat, time, sx, sy);
-		}
-	} else {
-		/*
-		 * Clear pointer focus so future button events and such are not
-		 * sent to the last client to have the cursor over it.
-		 *
-		 * Except if we started pressing a button on the last client and
-		 * are not currently in drag and drop mode.
-		 */
-		wlr_seat_pointer_clear_focus(wlr_seat);
-	}
+	cursor_update_common(server, node, surface, sx, sy, view_area, time,
+		/*force_reenter*/ false, /*cursor_has_moved*/ true);
```

Comment:
I love these comments. Will do that myself from now on.

Reply:
C's version of named parameters, haha